app/controllers/admin/api_docs/account_api_docs_controller.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class Admin::ApiDocs::AccountApiDocsController < Admin::ApiDocs::BaseController
+  before_action :redirect_to_service_scope, if: :service, only: %i[edit preview]
+
+  private
+
+  def redirect_to_service_scope
+    case action_name.to_sym
+    when :preview
+      redirect_to preview_admin_service_api_doc_path(service, api_docs_service)
+    when :edit
+      redirect_to edit_admin_service_api_doc_path(service, api_docs_service)
+    else
+      raise 'Not Implemented'
```

Comment:
This should not throw an 500 to the users.
Rather log it in System error reporting and use 404
Better to reuse a tagged error. I do not know if `ActionController::RoutingError` works with 404

Reply:
I don't think that is a good idea @hallelujah ...
a 4XX error should mean the user messed up somehow, and a 5XX that the developers/ops messed up. In this case, this can never happen for anything the user does... it can only happen if we change the previous `before_action :redirect_to_service_scope, if: :service, only: %i[edit preview]` to add any other action in the `only` but we don't add it here... so it is indeed a `Not Implemented error` and an error of the developer 😄 just like when we do in a `Base` class a `Not Implemented` if the developer doesn't override that method 😄

Reply:
Sure but it should still not throw a 500 to the user
That’s ugly

Reply:
Then it should never throw a 500 even in the `Inherited < Base` when something is not implemented, no? And why do we currently do it like that everywhere then?

Reply:
Like for example why is it acceptable to raise 500 here then?:
https://github.com/3scale/porta/pull/72/files#diff-951c55742156f26f8a9143ca15278719R93
It should be the same, no? Either we consider 5XX to the user acceptable or we don't 😕

Reply:
Well in this case if someone manually forge the URL with an invalid action what happens?
I mean it can be the developer fault OR the user fault OR both
So you do not control it
In the other case it is the developer fault only

Reply:
And why we don't add the `NotImplementedError` in here then?:
https://github.com/3scale/porta/blob/6530929a6226e5dad3cb17394f54e3a4447233c3/app/controllers/application_controller.rb#L47-L65

Reply:
Because the layer it was raised might not be in controller but deeper
Like in a service or in a model